src/ui/Base/ComponentOptionsLoader.ts

```diff
@@ -0,0 +1,85 @@
+import { extend, find } from 'underscore';
+import { Logger } from '../../misc/Logger';
+import { Utils } from '../../utils/Utils';
+import { ComponentOptionsType, IComponentLocalizedStringOptionArgs, IComponentOptionsOption } from './IComponentOptions';
+
+export class ComponentOptionLoader {
+  private resolvedValue: any = null;
+  private logger: Logger;
+
+  constructor(
+    public element: HTMLElement,
+    public values: Record<any, any>,
+    public optionName: string,
+    public optionDefinition: IComponentOptionsOption<any>
+  ) {
+    this.logger = new Logger(this);
+  }
+
+  public load() {
+    return this.findFirstValidValue(
+      this.loadFromAttribute.bind(this),
+      this.loadFromOptionsDictionnary.bind(this),
+      this.loadFromDefaultValue.bind(this),
+      this.loadFromDefaultFunction.bind(this)
+    );
+  }
+
+  private loadFromAttribute() {
+    return this.optionDefinition.load ? this.optionDefinition.load(this.element, this.optionName, this.optionDefinition) : null;
+  }
+
+  private loadFromOptionsDictionnary() {
+    return this.values[this.optionName];
+  }
+
+  private loadFromDefaultValue() {
+    if (this.optionDefinition.type == ComponentOptionsType.LOCALIZED_STRING) {
+      return this.loadDefaultLocalizedString();
+    }
+
+    if (Utils.isNullOrUndefined(this.optionDefinition.defaultValue)) {
+      return null;
+    }
+
+    switch (this.optionDefinition.type) {
+      case ComponentOptionsType.LIST:
+        return extend([], this.optionDefinition.defaultValue);
+      case ComponentOptionsType.OBJECT:
+        return extend({}, this.optionDefinition.defaultValue);
```

Comment:
Could you just use ES6 instead of extend to copy the list/object
[...this.optionDefinition.defaultValue]
{...this.optionDefinition.defaultValue}